File: stack/Solution71.java
import java.util.Stack;

class Solution71 {
    public String simplifyPath(String path) {
		Stack<String> stack = new Stack<>();
        for(String s: path.split("/")){
            switch(s){
                case "..":
                if(!stack.empty()){
                    stack.pop();
                }
                break;

                case ".":
                break;

                case "":
                break;

                default:
                stack.push(s);
                break;
            }
        }

        String result = "";
        if(stack.empty()){
            return "/";
        }

        while(!stack.empty()){
            result = "/" + stack.pop() + result;
        }
        return result;
    }

    public static void main(String[] args){
    	Solution71 s = new Solution71();
    	String s1 = "/home/";
    	System.out.println(s.simplifyPath(s1));
    	String s2 = "/../";
		System.out.println(s.simplifyPath(s2));
		String s3 = "/home//foo/";
		System.out.println(s.simplifyPath(s3));
		String s4 = "/a/./b/../../c/";
		System.out.println(s.simplifyPath(s4));
		String s5 ="/a/../../b/../c//.//";
		System.out.println(s.simplifyPath(s5));
		String s6 = "/a//b////c/d//././/..";
		System.out.println(s.simplifyPath(s6));
		String s7 = "/...";
		System.out.println(s.simplifyPath(s7));
    }
}

File: C++/Solution71.cpp
#include<iostream>
#include<vector>
#include<algorithm>
#include<unordered_map>
#include<unordered_set>
#include<stack>
using namespace std;

class Solution71 {
public:
    string simplifyPath(string path) {
        string res = "";
        stack<string> stk;
        vector<string> path2 = split(path, '/');
        for(string str: path2){
            if(str == "."){
                continue;
            }
            if(str == ".."){
                if(!stk.empty()){
                    stk.pop();
                }
            }
            else{
                stk.push(str);
            }
        }
        if(stk.empty()){
            return "/";
        }
        while(!stk.empty()){
            res = stk.top() + res;
            res = "/" + res;
            stk.pop();  
        }
        return res;
    }

    vector<string> split(string path, char x){
        vector<string> res;
        int start = 0;
        while(start < path.length()){
            while(start < path.length() && path[start] == x){
                start++;
            }
            if(start == path.length()){
                return res;
            }
            int end = start;
            while(end < path.length() && path[end] != x){
                end++;
            }
            end--;
            res.push_back(path.substr(start, end - start + 1));
            start = end + 1;
        }
        return res;
    }
};

int main(){
    Solution71 s;
    string str = "/hom/////e/";
    vector<string> test = s.split(str, '/');
    for(auto ele: test){
        cout << ele << " ";
    }
}